controllers/helmrelease_controller_chart.go

```diff
@@ -240,6 +245,10 @@ func helmChartRequiresUpdate(hr *v2.HelmRelease, chart *sourcev1.HelmChart) bool
 		return true
 	case template.Spec.ValuesFile != chart.Spec.ValuesFile:
 		return true
+	case !reflect.DeepEqual(template.Annotations, chart.Annotations):
+		return true
+	case !reflect.DeepEqual(template.Labels, chart.Labels):
+		return true
```

Comment:
Maps have a random order, and the comparison must thus be semantic. See e.g.

https://github.com/fluxcd/pkg/blob/main/ssa/manager_diff.go#L151